backend/cluster/cluster_crud.go

```diff
@@ -0,0 +1,192 @@
+package cluster
```

Comment:
General comment:
There is a ton of repetition in this file - details discussed on call